### PR TITLE
Add evaluation module, from_mot conversion script

### PR DESCRIPTION
**Pull request by tobiasfshr**
Add evaluation scripts for detect / mot. Add a conversion script to convert MOTChallenge datasets to Scalabel format.

### CODE REVIEW

**Review by XiaLiPKU — DISMISSED** [review]

**Review by XiaLiPKU — DISMISSED** [review]

File: scalabel/common/typing.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 """Common types used in the project."""
 
-from typing import Any, Dict
+from typing import Any, Dict, List
 
 DictStrAny = Dict[str, Any]  # type: ignore[misc]
+DictAny = Dict[str, Any]  # type: ignore[misc]
```
Comment (fyu) [inline scalabel/common/typing.py:6]: This is not necessary

**Review by XiaLiPKU — APPROVED** [review]

File: scalabel/eval/detect.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    Args:
+        small_dict (dict): a result dictionary of only a few items.
+
+    Returns:
+        str: the table as a string.
+    """
+    keys, values_t = tuple(zip(*small_dict.items()))
+    values = ["{:.1f}".format(val * 100) for val in values_t]
+    stride = 3
+    items: ListAny = []
```
Comment (fyu) [inline scalabel/eval/detect.py:228]: This use of Any is avoidable since we know the types in `small_dict`.

File: scalabel/eval/detect.py [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+        "--gt", "-g", required=True, help="path to detection ground truth"
+    )
+    parser.add_argument(
+        "--result", "-r", required=True, help="path to detection results"
+    )
+    parser.add_argument(
+        "--cfg-path",
+        "-c",
+        default=DEFAULT_LABEL_CONFIG,
+        help="Config path, contains metadata like available categories.",
```
Comment (fyu) [inline scalabel/eval/detect.py:256]: Can we be more precise for the config schema?

File: scalabel/eval/detect.py [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+    """Parse the arguments."""
+    parser = argparse.ArgumentParser(description="Detection evaluation.")
+    parser.add_argument(
+        "--gt", "-g", required=True, help="path to detection ground truth"
+    )
+    parser.add_argument(
+        "--result", "-r", required=True, help="path to detection results"
+    )
+    parser.add_argument(
+        "--cfg-path",
```
Comment (fyu) [inline scalabel/eval/detect.py:253]:
```suggestion
        "--config",
```

File: scalabel/label/typing.py [hunk 5]
```diff
@@ -94,3 +93,37 @@ def __init__(self, **data: Any) -> None:  # type: ignore
...
+
+Category.update_forward_refs()
+
+
+class Attribute(BaseModel):
+    """Define Scalabel category type."""
+
+    name: str
+    toolType: str
+    tagText: str
```
Comment (fyu) [inline scalabel/label/typing.py:113]:
```suggestion
    type: str
    tag: str
```

File: scalabel/label/typing.py [hunk 6]
```diff
@@ -94,3 +93,37 @@ def __init__(self, **data: Any) -> None:  # type: ignore
...
+    values: Optional[List[str]]
+
+
+class Config(BaseModel):
+    """Define metadata of the dataset."""
+
+    resolution: Optional[Tuple[int, int]]
```
Comment (fyu) [inline scalabel/label/typing.py:120]: a `Size` class will be easier to understand here.

File: scalabel/eval/detect.py (hunk 2 above)
Comment (tobiasfshr) [reply scalabel/eval/detect.py]: done

File: scalabel/eval/detect.py (hunk 3 above)
Comment (tobiasfshr) [reply scalabel/eval/detect.py]: made more concrete

File: scalabel/eval/detect.py (hunk 4 above)
Comment (tobiasfshr) [reply scalabel/eval/detect.py]: changed

File: scalabel/label/typing.py (hunk 6 above)
Comment (tobiasfshr) [reply scalabel/label/typing.py]: done

**Pull request merged**